### ISSUE TITLE
Build-time failure at POSIX.so on threaded builds on FreeBSD and NetBSD

### ISSUE THREAD

**Issue by jkeenan** [Perl/perl5#18391]
On threaded builds on FreeBSD-11, I am currently experiencing a build-time failure as follows:
```
./miniperl -Ilib make_ext.pl lib/auto/POSIX/POSIX.so  MAKE="make" LIBPERL_A=libperl.a LINKTYPE=dynamic
Warning (mostly harmless): No library found for -lposix
Warning (mostly harmless): No library found for -lcposix
Generating a Unix-style Makefile
Writing Makefile for POSIX
Running Mkbootstrap for POSIX ()
chmod 644 "POSIX.bs"
/usr/home/jkeenan/gitwork/perl/ext/POSIX/../../miniperl "-I../../lib" -MExtUtils::Command::MM -e 'cp_nonempty' -- POSIX.bs ../../lib/auto/POSIX/POSIX.bs 644
"../../miniperl" "-I../../lib" "../../lib/ExtUtils/xsubpp" -noprototypes -typemap '/usr/home/jkeenan/gitwork/perl/ext/POSIX/../../lib/ExtUtils/typemap' -typemap '/usr/home/jkeenan/gitwork/perl/ext/POSIX/typemap'  POSIX.xs > POSIX.xsc
mv POSIX.xsc POSIX.c
cc -c    -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -DSTRUCT_TM_HASZONE -Wall -Werror=pointer-arith -Wextra -Wc++-compat -Wwrite-strings -Werror=declaration-after-statement -Wthread-safety -O2 -pipe -fstack-protector -fno-strict-aliasing    -DVERSION=\"1.96\"  -DXS_VERSION=\"1.96\" -DPIC -fPIC "-I../.."   POSIX.c
POSIX.xs:2188:9: error: use of undeclared identifier 'LOCALECONV_LOCK'
        LOCALECONV_LOCK;    /* Prevent interference with other threads using
        ^
POSIX.xs:2274:9: error: use of undeclared identifier 'LOCALECONV_UNLOCK'
        LOCALECONV_UNLOCK;
        ^
2 errors generated.
*** Error code 1

Stop.
make[1]: stopped in /usr/home/jkeenan/gitwork/perl/ext/POSIX
/usr/home/jkeenan/gitwork/perl/ext/POSIX/../../miniperl "-I../../lib" -MExtUtils::Command::MM -e 'cp_nonempty' -- POSIX.bs ../../lib/auto/POSIX/POSIX.bs 644
cc -c    -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -DSTRUCT_TM_HASZONE -Wall -Werror=pointer-arith -Wextra -Wc++-compat -Wwrite-strings -Werror=declaration-after-statement -Wthread-safety -O2 -pipe -fstack-protector -fno-strict-aliasing    -DVERSION=\"1.96\"  -DXS_VERSION=\"1.96\" -DPIC -fPIC "-I../.."   POSIX.c
POSIX.xs:2188:9: error: use of undeclared identifier 'LOCALECONV_LOCK'
        LOCALECONV_LOCK;    /* Prevent interference with other threads using
        ^
POSIX.xs:2274:9: error: use of undeclared identifier 'LOCALECONV_UNLOCK'
        LOCALECONV_UNLOCK;
        ^
2 errors generated.
*** Error code 1

Stop.
make[1]: stopped in /usr/home/jkeenan/gitwork/perl/ext/POSIX
Unsuccessful make(ext/POSIX): code=256 at make_ext.pl line 588.
*** Error code 25

Stop.
make: stopped in /usr/home/jkeenan/gitwork/perl
```
Commit: f4ed6d050b

C-compiler:
$ clang --version
FreeBSD clang version 6.0.1 (tags/RELEASE_601/final 335540) (based on LLVM 6.0.1)
Target: x86_64-unknown-freebsd11.2
Thread model: posix
InstalledDir: /usr/bin

$ uname -mrs
FreeBSD 11.2-STABLE amd64

I am not experiencing this on unthreaded builds on this machine.  Both unthreaded and threaded builds on Linux are not affected.  I am not sure at which commit this failure first appeared.  The last time I tried to build blead on this machine was at commit 0aee951734 on Dec 02.

**Comment by jkeenan** [comment]
Also getting this problem on threaded builds on NetBSD (though not on threaded builds on OpenBSD):
```
./miniperl -Ilib make_ext.pl lib/auto/POSIX/POSIX.so  MAKE="make" LIBPERL_A=libperl.a LINKTYPE=dynamic
Warning (mostly harmless): No library found for -lcposix
Generating a Unix-style Makefile
Writing Makefile for POSIX
Running Mkbootstrap for POSIX ()
chmod 644 "POSIX.bs"
/home/jkeenan/gitwork/perl/ext/POSIX/../../miniperl "-I../../lib" -MExtUtils::Command::MM -e 'cp_nonempty' -- POSIX.bs ../../lib/auto/POSIX/POSIX.bs 644
"../../miniperl" "-I../../lib" "../../lib/ExtUtils/xsubpp" -noprototypes -typemap '/home/jkeenan/gitwork/perl/ext/POSIX/../../lib/ExtUtils/typemap' -typemap '/home/jkeenan/gitwork/perl/ext/POSIX/typemap'  POSIX.xs > POSIX.xsc
mv POSIX.xsc POSIX.c
cc -c    -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/pkg/include -D_FORTIFY_SOURCE=2 -DSTRUCT_TM_HASZONE -Wall -Werror=pointer-arith -Wextra -Wc++-compat -Wwrite-strings -Werror=declaration-after-statement -O    -DVERSION=\"1.96\"  -DXS_VERSION=\"1.96\" -DPIC -fPIC "-I../.."   POSIX.c
POSIX.xs: In function 'XS_POSIX_localeconv':
POSIX.xs:2188:9: error: 'LOCALECONV_LOCK' undeclared (first use in this function)
         LOCALECONV_LOCK;    /* Prevent interference with other threads using
         ^
POSIX.xs:2188:9: note: each undeclared identifier is reported only once for each function it appears in
POSIX.xs:2274:9: error: 'LOCALECONV_UNLOCK' undeclared (first use in this function)
         LOCALECONV_UNLOCK;
         ^
*** Error code 1

Stop.
make[1]: stopped in /home/jkeenan/gitwork/perl/ext/POSIX
/home/jkeenan/gitwork/perl/ext/POSIX/../../miniperl "-I../../lib" -MExtUtils::Command::MM -e 'cp_nonempty' -- POSIX.bs ../../lib/auto/POSIX/POSIX.bs 644
cc -c    -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/pkg/include -D_FORTIFY_SOURCE=2 -DSTRUCT_TM_HASZONE -Wall -Werror=pointer-arith -Wextra -Wc++-compat -Wwrite-strings -Werror=declaration-after-statement -O    -DVERSION=\"1.96\"  -DXS_VERSION=\"1.96\" -DPIC -fPIC "-I../.."   POSIX.c
POSIX.xs: In function 'XS_POSIX_localeconv':
POSIX.xs:2188:9: error: 'LOCALECONV_LOCK' undeclared (first use in this function)
         LOCALECONV_LOCK;    /* Prevent interference with other threads using
         ^
POSIX.xs:2188:9: note: each undeclared identifier is reported only once for each function it appears in
POSIX.xs:2274:9: error: 'LOCALECONV_UNLOCK' undeclared (first use in this function)
         LOCALECONV_UNLOCK;
         ^
*** Error code 1

Stop.
make[1]: stopped in /home/jkeenan/gitwork/perl/ext/POSIX
Unsuccessful make(ext/POSIX): code=256 at make_ext.pl line 588.
*** Error code 25

Stop.
make: stopped in /home/jkeenan/gitwork/perl
```
Configure:
```
sh ./Configure -des -Dusedevel \
            -Duseithreads \
            -Uusemymalloc
```
$ uname -mrs
NetBSD 8.0 amd64

C-compiler:
$ gcc --version
gcc (nb2 20180327) 5.5.0

**Comment by jkeenan** [comment]
Appears to have been fixed by c07463d862e4832cc6a94200a77a3170ef2dca18.

Thank you very much.
Jim Keenan
